test/test_jit.py

```diff
@@ -3352,23 +3362,38 @@ def func():
         # node hashing correctly working, no CSE occurs
         self.assertEqual(len(constants) + 1, str(f_script.graph).count("prim::Constant"))
 
+
+        # t is used to test CSE against
+        # dicts with tensors as keys
         funcs_template = dedent('''
         def func():
+            t = torch.tensor(42.)
             a = {constant_constructor}
-            print(a)
+            print(a, t)
             b = {constant_constructor}
-            print(b)
+            print(b, t)
         ''')
 
-        # test that equal tuples correctly work with node hashing
-        for tup in map(lambda x: "(" + x + ",)", constants):
+        # generate dicts with built-in types (excluding torch.Tensor)
+        xprod = itertools.product(constants, constants)
+
+        def keys_pred(t):
+            return isinstance(eval(t[0]), (list, bool, Tensor)) or eval(t[0]) is None
+
+        filt = [x for x in xprod if not keys_pred(x)]
+        prim_dict_strs = map(lambda t: '{' + t[0] + ':' + t[1] + '}', filt)
+        ten_dict_strs = ['{t : 0}', '{42. : t}']
+        dict_strs = chain(prim_dict_strs, ten_dict_strs)
+
+        # test that equal tuples and dicts correctly work with node hashing
+        for tup in chain(map(lambda x: "(" + x + ",)", constants), dict_strs):
             funcs_str = funcs_template.format(constant_constructor=tup)
             scope = {}
             execWrapper(funcs_str, globals(), scope)
             cu = torch.jit.CompilationUnit(funcs_str)
             f_script = cu.func
             self.run_pass('constant_propagation', f_script.graph)
-            num_constants = 3  # input constant twice, None once
+            num_constants = 4  # input constant twice, None once
```

Comment:
Could you update the comment here? `# input constant twice, None once`

Reply:
will do